fastlane/Fastfile

```diff
@@ -239,7 +262,80 @@ end
       project_slug: 'simplenote-macos',
       dsym_path: dSYM_PATH,
     )
+  end
+
+  #############################################################################
+  # build_and_upload_app_store
+  # ---------------------------------------------------------------------------
+  # This lane builds the app and uploads it to the Mac App Store, optionally
+  # creating a GitHub release.
+  # ---------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane build_and_upload_app_store [create_github_release:<create github release>]
+  #
+  # Example:
+  # bundle exec fastlane build_and_upload_app_store
+  # bundle exec fastlane build_and_upload_app_store create_github_release:true
+  #############################################################################
+  lane :build_and_upload_app_store do | options |
+    # This allows code signing to work on CircleCI. It is skipped if this isn't
+    # running on CI.
+    #
+    # See https://circleci.com/docs/2.0/ios-codesigning/
+    setup_circle_ci
+
+    sync_code_signing(
+      type: 'appstore',
+      platform: 'macos',
+      # This Mac app also needs a Mac Installer Distribution certificate
+      additional_cert_types: ['mac_installer_distribution'],
+      # Will default to false, unless a readonly value is specified, or is
+      # running on CI, in which case it will always be true.
+      readonly: options[:readonly] || is_ci,
+      app_identifier: 'com.automattic.SimplenoteMac'
+    )
+
+    archive_path = File.join(BUILD_FOLDER, 'Simplenote-Mac.xcarchive')
+    build_simplenote(codesign: true, archive_path: archive_path)
```

Comment:
This is where I was kinda expecting for you to call `build_app_store` until I realized the true intent of that other rule :P